src/libraries/System.Net.WebProxy/tests/WebProxyTest.cs

```diff
@@ -132,6 +132,27 @@ public static void WebProxy_InvalidBypassUrl_AddedDirectlyToList_SilentlyEaten()
             p.IsBypassed(new Uri("http://microsoft.com")); // exception should be silently eaten
         }
 
+        [Fact]
+        public static void WebProxy_BypassUrl_AddedDirectlyToList_IsBypassed()
+        {
+            var p = new WebProxy("http://microsoft.com", false);
+            Assert.Equal(new Uri("http://microsoft.com"), p.GetProxy(new Uri("http://bing.com")));
+            p.BypassArrayList.Add("bing");
+            Assert.Equal(new Uri("http://bing.com"), p.GetProxy(new Uri("http://bing.com")));
+        }
+
+        [Fact]
+        public static void WebProxy_BypassRegexList_ShouldNotRefreshBetweenCalls()
+        {
+            var p = new WebProxy("http://microsoft.com", false, new string[] { "foobar" });
+
+            //not sure what the policy is for reflexion in unit tests...
```

Comment:
We try to avoid it.

I don't think we need to test that instance isn't being regenerated.  I'd rather see tests that all possible updates to the ArrayList are reflected on subsequent calls, e.g. you've added explicit code to set IsChanged when regexes are added/removed/inserted/changed/etc... we should have tests then validating that all possible mutations on the returned ArrayList are reflected in subsequent IsBypassed calls.